core/services/offchainreporting/peer_wrapper.go

```diff
@@ -0,0 +1,126 @@
+package offchainreporting
+
+import (
+	"sync"
+
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/store/orm"
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting/types"
+)
+
+type (
+	peer interface {
+		ocrtypes.BootstrapperFactory
+		ocrtypes.BinaryNetworkEndpointFactory
+	}
+
+	// SingletonPeerWrapper manages all libocr peers for the application
+	SingletonPeerWrapper struct {
+		keyStore *KeyStore
+		config   *orm.Config
+		db       *gorm.DB
+
+		pstoreWrapper *Pstorewrapper
+		PeerID        models.PeerID
+		Peer          peer
+
+		startMu *sync.Mutex
+		started bool
+	}
+)
+
+// NewSingletonPeerWrapper creates a new peer based on the p2p keys in the keystore
+// It currently only supports one peerID/key
+// It should be fairly easy to modify it to support multiple peerIDs/keys using e.g. a map
+func NewSingletonPeerWrapper(keyStore *KeyStore, config *orm.Config, db *gorm.DB) *SingletonPeerWrapper {
+	return &SingletonPeerWrapper{keyStore, config, db, nil, "", nil, new(sync.Mutex), false}
+}
+
+func (p *SingletonPeerWrapper) IsStarted() bool {
+	p.startMu.Lock()
+	defer p.startMu.Unlock()
+	return p.started
+}
+
+func (p *SingletonPeerWrapper) Start() (err error) {
+	p.startMu.Lock()
+	defer p.startMu.Unlock()
+
+	if p.started {
+		return errors.New("already started")
+	}
+
+	p.started = true
+
+	p2pkeys := p.keyStore.DecryptedP2PKeys()
+	listenPort := p.config.P2PListenPort()
+	if listenPort == 0 {
+		return errors.New("failed to instantiate oracle or bootstrapper service, P2P_LISTEN_PORT is required and must be set to a non-zero value")
+	}
+
+	if len(p2pkeys) == 0 {
+		return nil
+	} else if len(p2pkeys) > 1 {
+		return errors.New("more than one p2p key is not currently supported")
+	} else {
```

Comment:
nit: since the above two branches return, we can just chop this else and avoid the indentation here